lib/puma/server.rb

```diff
@@ -504,22 +504,22 @@ def with_force_shutdown(client, &block)
     # :nocov:
 
     # Handle various error types thrown by Client I/O operations.
-    def client_error(e, client)
+    def client_error(e, client, requests = 1)
       # Swallow, do not log
       return if [ConnectionError, EOFError].include?(e.class)
 
-      lowlevel_error(e, client.env)
```

Comment:
Why does MiniSSL::SSLError get slightly different handling now? If I'm reading it right, this branch still has the "bug" of never responding to the client actually?

Reply:
Adding `lowlevel_error` to `MiniSSL::SSLError` so that the handler code can be run, and we don't send anything back to client because there is nothing to send back right? If I understand your previous comment correctly